src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -304,6 +291,68 @@ impl LogicalAgg {
         }
     }
 
+    /// Prepares metadata and the `approx_percentile` plan, if there's one present.
+    fn prepare_approx_percentile(
```

Comment:
This function would modify the `core.agg_calls and core.input`. We'd better document this behavior.